packages/custom_lint_builder/lib/src/client.dart

```diff
@@ -374,7 +374,9 @@ class _ClientAnalyzerPlugin extends analyzer_plugin.ServerPlugin {
       final pubspecs = {
         for (final analysisContext in contextCollection.contexts)
           analysisContext: parsePubspec(
-            io.Directory(analysisContext.contextRoot.root.path),
+            findProjectDirectory(
+              io.Directory(analysisContext.contextRoot.root.path),
+            ),
```

Comment:
I was personally thinking of doing something among the lines of:

```dart
final projectDirectory = findPRojectDir();
if (projectDirectory == null) continue;
```

I don't quite see the purpose of throwing here.

Reply:
The if case expression was causing issues, I'm not sure how analysis plugins and language versions work. I updates the pubspecs to dart 3.0.0, but can revert those changes if you'd like.

Reply:
I'm fine with bumping the minimum sdk to 3.0.0
I've done that in a branch too anyway.

On the other hand, the issue about not throwing hasn't been resolved. Is that what you meant by "causing issues"? If so, what were the problems?